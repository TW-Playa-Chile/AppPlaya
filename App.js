import React from "react";
import {Alert, AsyncStorage, Button, TextInput, View} from "react-native";

const bind = (context) => (...methods) => (methods.forEach(method => context[method] = context[method].bind(context)));

export default class App extends React.Component {
    constructor(props) {
        super(props);
        this.state = {text: ''};
        bind(this)('_onPressSave', '_onPressClear');
    }

    async _onPressSave() {
        try {
            let activityText = this.state.text.trim();
            await AsyncStorage.getItem('@ActivityStore:activities', (err, result) => {
                let allActivities = [];
                let storedActivities = JSON.parse(result);
                if (storedActivities) {
                    allActivities = storedActivities;
                }
                AsyncStorage.setItem('@ActivityStore:activities', JSON.stringify(allActivities.concat([activityText])));
            });
            await AsyncStorage.getItem('@ActivityStore:activities', (err, result) => {
                Alert.alert('Registro exitoso', 'actividades: ' + result);
            });
        } catch (error) {
            console.log(error);
            Alert.alert('Error en el registro');
        }
        this.setState({text: ''});
    }

    async _onPressClear() {
        await AsyncStorage.clear()
            .then(Alert.alert('Cleared'))
            .catch(error => Alert.alert('Error'));
    }

    render() {
        return (
            <View style={{flex: 1, justifyContent: 'center'}}>
                <View style={{padding: 10, borderBottomColor: '#000000', borderBottomWidth: 1 }}>
                    <TextInput
                        underlineColorAndroid={'transparent'}
                        onChangeText={(text) => this.setState({text})}
                        value={this.state.text}

                    />
                </View>
                <View style={{padding: 10}}>
                    <Button
                        onPress={this._onPressSave}
                        title="Save"
                        color="#841584"
                        accessibilityLabel="Save new activity"
                    />
                </View>
                <View style={{padding: 10}}>
                    <Button
                        onPress={this._onPressClear}
                        title="Clear"
                        color="#841584"
                        accessibilityLabel="Remove all activity"
                    />
                </View>
            </View>
        );
    }
}
