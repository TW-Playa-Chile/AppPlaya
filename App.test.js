import 'react-native';
import React from 'react';
import App from './App';
import { shallow } from 'enzyme';
import {TextInput} from 'react-native';

import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer.create(
      <App />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

it('should change the activity text state when input is changed', () => {
    const fakeActivity = 'Activity 1';
    const wrapper = shallow(<App/>);

    wrapper.find(TextInput).simulate('ChangeText',fakeActivity);

    expect(wrapper.state('text')).toBe(fakeActivity);
});