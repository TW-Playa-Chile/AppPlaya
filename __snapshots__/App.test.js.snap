// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "borderBottomColor": "#000000",
        "borderBottomWidth": 1,
        "padding": 10,
      }
    }
  >
    <TextInput
      onChangeText={[Function]}
      underlineColorAndroid="transparent"
      value=""
    />
  </View>
  <View
    style={
      Object {
        "padding": 10,
      }
    }
  >
    <View
      accessibilityComponentType="button"
      accessibilityLabel="Save new activity"
      accessibilityTraits={
        Array [
          "button",
        ]
      }
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
              Object {
                "color": "#841584",
              },
            ]
          }
        >
          Save
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "padding": 10,
      }
    }
  >
    <View
      accessibilityComponentType="button"
      accessibilityLabel="Remove all activity"
      accessibilityTraits={
        Array [
          "button",
        ]
      }
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
              Object {
                "color": "#841584",
              },
            ]
          }
        >
          Clear
        </Text>
      </View>
    </View>
  </View>
</View>
`;
